### DISCUSSION TITLE
Custom JWT Auth Login Error

### DISCUSSION THREAD

**Discussion by 0xboga** [Web3Auth/Web3Auth#771] [Help]
Hey guys, i'm trying to implement the custom auth using JWT token. I've successfully deployed the verified in the dashboard but i get some weird errors. Here is a snippet of my code. Any clues on whats going wrong ?

`

    const web3authCore = new Web3AuthCore({
      chainConfig: {
        chainNamespace: CHAIN_NAMESPACES.EIP155,
        chainId: "0x38",
        rpcTarget: "https://bsc-dataseed1.binance.org",
        blockExplorer: "https://bscscan.com/",
        ticker: "BNB",
        tickerName: "Binance Smart Chain",
      },
      clientId: ""
    });

    const adapter = new OpenloginAdapter({ adapterSettings: {
      network: "testnet",
      clientId: "",
      uxMode: "popup",
      loginConfig: {
        jwt: {
          name: "any name",
          verifier: "wam-testnet",
          typeOfLogin: "jwt",
          clientId: "",
        },
      },
    }});

    web3authCore.configureAdapter(adapter);
    await web3authCore.init();

    const provider = await web3authCore.connectTo(adapter.name, { 
      loginProvider: "jwt",
      extraLoginOptions: {
        id_token: "",
        domain: "http://localhost:8080",
        verifierIdField: "sub", 
        response_type: "token",
        scope:"",
      }
    });

`
![Screenshot 2022-10-25 at 15 55 09](https://user-images.githubusercontent.com/35598195/197781867-03df7e0e-09df-42ad-afc2-1649c095815d.png)
![Screenshot 2022-10-25 at 16 35 28](https://user-images.githubusercontent.com/35598195/197792962-e75eaa1c-02c7-4be3-a5bd-b5966bfd1aab.png)

**Comment by himanshuchawla009** [comment]
Can you show other error logs in red, you can ignore yellow colored warnings.

**Comment by 0xboga** [comment]
<img width="1048" alt="Screenshot 2022-10-25 at 20 36 15" src="https://user-images.githubusercontent.com/35598195/197843529-400e8bcb-c929-4d1c-aea4-4c79b7b5b772.png">

**Comment by shahbaz17** [comment]
Hey @bogdanpurnavel 

Can you share one of the tokens here? 

Also these details:
* JWKS endpoint
* JWT Validation fields

**Reply by 0xboga** [reply]
> Hey @bogdanpurnavel
> 
> Can you share one of the tokens here?
> 
> Also these details:
> 
> * JWKS endpoint
> * JWT Validation fields

- Token: eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IjA5MDRmMWE4LWFlODktNGYwZi05Mjc3LTFkYWI1ODI4NzU5MyJ9.eyJzdWIiOiI1NTI2MGM4NC03ZWE4LTRhMWUtYjZlOS1kNmI0NWJhNzMwMDQiLCJuYW1lIjoidmlvcmVsQGRpZ2l0YXAuZXUiLCJhdWQiOiJ1cm46MDVhNC01LTItMjMwLTI0Mi5uZ3Jvay5pbyIsImlzcyI6Imh0dHBzOi8vMDVhNC01LTItMjMwLTI0Mi5uZ3Jvay5pbyIsImlhdCI6MTY2Njc2ODE4OSwiZXhwIjoxNjY3MzcyOTc3fQ.tLvdwm1ClWsCeVx4sZGUY0C7h89tSCtKZrhWrTyhyM8xlAJD113ygr22w8Yh4igiS3Uo_b_pfxzw8euAp_SONGw2uPr2UcRorFuLbO8fWwgaysa5YMo0zJA6RenpUT8fVpJ9RRFhs_gDBabI8PSfTO-nUXJxF6EWJDBtC9cHceE6qNvN7sHaEask19plsdeYHCIN04E_VJgJ-Hemp-9neyoae1NHubjc0_Z53TPZl6NNmkkv7vsbJEaWIy11ORSo0Ar4eA2DhWRwNdzQ1pSoY32QG5T3GMLfur4QI0GclLWVsGh6KfkUzZN9-Oz1lYn5h5Lcex0oaAn0wjiMg3kArw
- JWKS endpoint: https://05a4-5-2-230-242.ngrok.io/.well-known/jwks.json
- JWT Validation fields from verifier:
         - aud: urn:05a4-5-2-230-242.ngrok.io
         - iss: https://05a4-5-2-230-242.ngrok.io

**Reply by shahbaz17** [reply]
Unable to reach https://05a4-5-2-230-242.ngrok.io/.well-known/jwks.json

<img width="1127" alt="Screenshot 2022-10-26 at 2 20 24 PM" src="https://user-images.githubusercontent.com/6962565/197980578-c7e07f9d-1e94-434d-ac47-872b713a30f1.png">

**Reply by 0xboga** [reply]
hmm, can u try again now ? it should be working

<img width="1428" alt="Screenshot 2022-10-26 at 11 55 26" src="https://user-images.githubusercontent.com/35598195/197981924-777acf97-2f1f-4f44-b5de-63e4b33828a7.png">

**Comment by 0xboga** [comment]
Solved. issue was using a wrong verifier.
